### PR TITLE
reduce complexity of ManagedCluster spec

### LINKED ISSUE #3810: Reduce complexity of `Parameters()` func
/kind bug

**What steps did you take and what happened:**

The [`ManagedClusterSpec.Parameters()`](https://github.com/pluralsh/cluster-api-provider-azure/blob/64a02e09172a512c0f8630907be15f55ca49e0da/azure/services/managedclusters/spec.go#L270) func is long, with many `if`s and `nil` checks, so it currently exceeds the maximum cyclomatic complexity allowed by the `gocyclo` linter. We've disabled that linter for this func.

**What did you expect to happen:**

The func could be refactored to reduce [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity), perhaps by breaking it into multiple funcs. Existing unit test coverage should be preserved (or improved) across the refactor, and the `//nolint:gocyclo` comment should be removed.

**Anything else you would like to add:**

You can test locally that your refactoring succeeds with `make lint`.

**Environment:**

- cluster-api-provider-azure version: `main`
- Kubernetes version: (use `kubectl version`): N/A
- OS (e.g. from `/etc/os-release`): N/A

### PR DESCRIPTION
**Pull request by bravebeaver**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

to reduce the complexity of the Params method which exceeds golangci-lint limits.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3810 

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [X] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by bravebeaver** [comment]
try to follow the pattern to build parts of managedCluster. some properties are left inside the main Params() method if they are 
- trivial (e.g., spec.name)
- too complicated (e.g., agentpool profile)


`func (s *spec) buildxxx(managedCluster)
`

**Comment by mboersma** [comment]
Thanks @bravebeaver, this turned out to be a lot of work! I think your approach to refactoring is reasonable.

Apparently the `TestParameters/no_existing_managed_cluster` unit test is still failing though. Please take a look whenever time permits.

**Comment by bravebeaver** [comment]
@mboersma will do. it seems to be a trait of refactoring that apparently nothing is changed but a lot has been changed.. 😅.

File: azure/services/managedclusters/spec.go [hunk 1]
```diff
@@ -721,21 +806,17 @@ func (s *ManagedClusterSpec) Parameters(ctx context.Context, existingObj genrunt
 	}
 
 	if s.Preview {
-		hub := &asocontainerservicev1hub.ManagedCluster{}
-		if err := managedCluster.ConvertTo(hub); err != nil {
-			return nil, err
-		}
-		prev := &asocontainerservicev1preview.ManagedCluster{}
-		if err := prev.ConvertFrom(hub); err != nil {
+		preview, err := convertV1ToPreview(managedCluster)
```
Comment (mboersma) [inline azure/services/managedclusters/spec.go:809]: Thank you for changing this: `prev` is an ambiguous variable name. 👍🏻

**Review by mboersma — COMMENTED** [review]
/lgtm

Thanks for doing this and my apologies for not reviewing sooner!

Could you squash the commits and rebase against `HEAD`?

**Pull request closed without merging**